client/app/scripts/superdesk/editor/editor.js

```diff
@@ -550,16 +550,375 @@ function EditorService(spellcheck, $rootScope, $timeout) {
     }
 }
 
-angular.module('superdesk.editor', ['superdesk.editor.spellcheck'])
+angular.module('superdesk.editor', ['superdesk.editor.spellcheck', 'angular-embed'])
 
     .service('editor', EditorService)
-
-    .directive('sdTextEditor', ['editor', 'spellcheck', '$timeout', function (editor, spellcheck, $timeout) {
-
-        var config = {
-            buttons: ['bold', 'italic', 'underline', 'quote', 'anchor'],
-            anchorInputPlaceholder: gettext('Paste or type a full link'),
-            disablePlaceholders: true,
+    .directive('sdAddEmbed', function() {
+        return {
+            scope: true,
+            require: ['sdAddEmbed', '^sdTextEditor'],
+            templateUrl: 'scripts/superdesk/editor/views/add-embed.html',
+            controllerAs: 'vm',
+            controller: ['embedService', '$element', '$timeout', '$q', 'lodash',
+                        function(embedService, $element, $timeout, $q, _) {
```

Comment:
It might make sense to define controller as a standalone function that is then just passed here as the value of the `controller` property.
It's a subject to personal preferences, though, I would personally find the directive definition more readable (also, inlining the controller needs a few extra levels of indentation, hindering readability on smaller screens).